app/javascript/controllers/file_auth_controller.js

```diff
@@ -259,4 +266,31 @@ export default class extends Controller {
   hideMessagePanel() {
     this.messagePanelTarget.hidden = true
   }
+
+  // To see if item is restricted by location, check the probe service json response
+  handleLocationRestricted(json, accessService) {
+    // The probe auth service is called for each file separately.
+    // If the location restriction target is available, then trigger auth denied message.
+    // The event will lead to the locked icon being displayed for this item
+    // This allows the lock window to show
+    const event = new CustomEvent('auth-denied', { accessService: accessService })
```

Comment:
I think we should stuff the location restricted message in this event.  Then we can have a controller devoted to displaying these messages, and this controller would not have to deal with this

Reply:
This can be a follow-on PR though.

Reply:
I think that approach makes sense and it would be fine to do in a follow up PR. @dnoneill and I were also discussing the other cases (embargo etc.) so it makes sense to have that logic in its own event/controller framework.